conductr_cli/resolvers/bintray_resolver.py

```diff
@@ -0,0 +1,153 @@
+        matching_versions = [attribute['values'][0]
+                             for attribute in attributes
+                             if attribute['type'] == 'version' and
+                             attribute['name'] == latest_compatibility_version and
+                             attribute['values']]
+        if not matching_versions:
+            return None
+        else:
+            matching_version = matching_versions[0]
+            matching_compatibility_version, matching_digest = matching_version.split('-')
+            return bintray_download_url(bintray_username, bintray_password, org, repo, package_name,
+                                        matching_compatibility_version, matching_digest)
+    else:
+        bintray_version = '{}-{}'.format(compatibility_version, digest)
+        files_endpoint = '{}/packages/{}/{}/{}/versions/{}/files'.format(BINTRAY_API_BASE_URL, org, repo, package_name,
+                                                                         bintray_version)
+        files = get_json(files_endpoint, bintray_username, bintray_password)
+        matching_files = [f
+                          for f in files
+                          if f['owner'] == org and
+                          f['repo'] == repo and
+                          f['package'] == package_name and
+                          f['version'] == bintray_version]
+        if len(matching_files) > 1:
+            raise BintrayResolutionError(
+                'Unable to resolve - multiple versions found for owner={} repo={} package={} version={}'.format(
+                    org, repo, package_name, bintray_version))
+        elif not matching_files:
+            raise BintrayResolutionError(
+                'Unable to find version for owner={} repo={} package={} version={}'.format(
+                    org, repo, package_name, bintray_version))
+        else:
+            path = matching_files[0]['path']
+            download_url = '{}/{}/{}/{}'.format(BINTRAY_DOWNLOAD_BASE_URL, org, repo, path)
+            return download_url
+
+
+def get_version_timestamp(bintray_username, bintray_password, org, repo, package_name, version):
+    version_endpoint = '{}/packages/{}/{}/{}/versions/{}'.format(BINTRAY_API_BASE_URL, org, repo, package_name, version)
+    version = get_json(version_endpoint, bintray_username, bintray_password)
+    return dateutil.parser.parse(version['updated'])
+
+
+def get_json(uri, username, password):
+    if username is not None and password is not None:
+        response = requests.get(uri, auth=(username, password))
+    else:
+        response = requests.get(uri)
+    response.raise_for_status()
+    return json.loads(response.text)
+
+
+def log(message, org, repo, package_name, compatibility_version, digest):
+    if compatibility_version is None and digest is None:
+        print('{} org={} repo={} bundle={} version={}'.format(message, org, repo, package_name, 'latest'))
```

Comment:
We should use the shorthand convention here i.e.

{org}/{repo}/{package_name}:latest

...same for the others below.